src/llamafactory/extras/misc.py

```diff
@@ -231,18 +231,31 @@ def torch_gc() -> None:
         torch.cuda.empty_cache()
 
 
-def try_download_model_from_ms(model_args: "ModelArguments") -> str:
-    if not use_modelscope() or os.path.exists(model_args.model_name_or_path):
+def try_download_model_from_other_hub(model_args: "ModelArguments") -> str:
+    if (not use_openmind() and not use_modelscope()) or os.path.exists(model_args.model_name_or_path):
         return model_args.model_name_or_path
 
-    try:
-        from modelscope import snapshot_download
+    if use_openmind():
+        try:
+            import openmind
+            from openmind.utils.hub import snapshot_download
 
-        revision = "master" if model_args.model_revision == "main" else model_args.model_revision
-        return snapshot_download(model_args.model_name_or_path, revision=revision, cache_dir=model_args.cache_dir)
-    except ImportError:
-        raise ImportError("Please install modelscope via `pip install modelscope -U`")
+            revision = "main" if model_args.model_revision == "main" else model_args.model_revision
+            return snapshot_download(model_args.model_name_or_path, revision=revision, cache_dir=model_args.cache_dir)
```

Comment:
`return snapshot_download(model_args.model_name_or_path, revision=model_args.model_revision, cache_dir=model_args.cache_dir)`

Reply:
Fixed, I overlooked that the default value of `model_args.model_revision` is `"main"`.